### ISSUE TITLE
Install wants to create /usr/local/share/jupyter ?!?

### ISSUE THREAD

**Issue by BjornFJohansson** [ipython-contrib/IPython-notebook-extensions#222]
Hi, I get the error below when trying to install IPython notebook extensions. How do I solve this?
I use Anaconda on Ubuntu 15.04 thanks /bjorn    

    In [5]: import IPython.html.nbextensions as nb
       ...: ext= 'https://github.com/ipython-contrib/IPython-notebook-extensions/archive/3.x.zip'
       ...: nb.install_nbextension(ext)
       ...: 
    Traceback (most recent call last):

      File "<ipython-input-5-a43e1688ef6a>", line 3, in <module>
        nb.install_nbextension(ext)

      File "/home/bjorn/anaconda/envs/bjorn/lib/python2.7/site-packages/IPython/html/nbextensions.py", line 176, in install_nbextension
        ensure_dir_exists(nbext)

      File "/home/bjorn/anaconda/envs/bjorn/lib/python2.7/site-packages/IPython/utils/path.py", line 588, in ensure_dir_exists
        os.makedirs(path, mode=mode)

      File "/home/bjorn/anaconda/envs/bjorn/lib/python2.7/os.py", line 150, in makedirs
        makedirs(head, mode)

      File "/home/bjorn/anaconda/envs/bjorn/lib/python2.7/os.py", line 157, in makedirs
        mkdir(name, mode)

    OSError: [Errno 13] Permission denied: '/usr/local/share/jupyter'

**Comment by damianavila** [comment]
http://ipython.org/ipython-doc/3/api/generated/IPython.html.nbextensions.html#IPython.html.nbextensions.install_nbextension

check `prefix` and see if it helps...

**Comment by JesterEE** [comment]
Or run those few lines of code in a python console with a login that can write to that location (e.g. sudo).

In a Terminal:
```
$ sudo python3
```

In the privileged Python3 console session:
```
>>> import IPython.html.nbextensions as nb
>>> ext = 'https://github.com/ipython-contrib/IPython-notebook-extensions/archive/3.x.zip'
>>> nb.install_nbextension(ext)
```

Now if you start-up the an un-privileged Jupyter notebook server, you will have those extensions installed.  Remember to follow the rest of the [3.x extension configuration](https://github.com/ipython-contrib/IPython-notebook-extensions/wiki/config-extension).  If you install for all users as specified above, you need to modify the nbextensions.py file as shown in my yet to be implemented [PR](https://github.com/ipython-contrib/IPython-notebook-extensions/pull/223/files).  Also, make sure to have pyyaml installed since it is a dependency of the extension configuration script.

```
$ sudo pip3 install pyyaml
```

Since this is an easy pitfall during installation on all platforms, might I be so bold as to recommend that the dev team update the install instructions with the contents of this discussion and/or update the install procedure to do this for the user.

**Comment by pybokeh** [comment]
@JesterEE +1  for "Since this is an easy pitfall during installation on all platforms, might I be so bold as to recommend that the dev team update the install instructions with the contents of this discussion and/or update the install procedure to do this for the user."  
  
I am having the same problem and I am not quite sure how to use the --prefix option and such.  So it would be nice if the installation could set the flag(s) right for the user.  The only way I have been able to install nb extensions successfully is using the ipython method:  
    
    import notebook
    E = notebook.nbextensions.EnableNBExtensionApp()
    E.enable_nbextension('usability/codefolding/main')  
  
I have my nb extensions installed at:
    
   /home/pybokeh/.local/share/jupyter folder

**Comment by jcb91** [comment]
This is yet another issue caused by our currently less-than-perfect install method.

another one fixed in #626

**Comment by juhasch** [comment]
Closing as outdated.
